In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/iris-flower-dataset/IRIS.csv


In [2]:
df = pd.read_csv('/kaggle/input/iris-flower-dataset/IRIS.csv')

Feature Matrix , $X$

In [3]:
# Feature Matrix , X
x = np.array(df.iloc[:,0:-1])
x[:5] # sample

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

In [4]:
# X dataset 
x.shape

(150, 4)

$X \text{ dataset consists of: }$
150 rows x 4 columns

$X$ is real number, $ X \in R^{150 x 4}$

$X_{1}^{150}$ is  first dimension of flower sample 150, the sepal length

Each row in this feature matrix represents one flower instance and can be written as four-dimensional row vector 



In [5]:
x[0]

array([5.1, 3.5, 1.4, 0.2])